bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -275,10 +274,10 @@ private Channel findChannel(String channelId, Thing thing) {
         return null;
     }
 
-    private void unlinkChannelIfAlreadyBound(ChannelUID channelUID) {
+    private void unlinkChannelIfAlreadyLinked(ChannelUID channelUID) {
         Collection<ItemChannelLink> links = managedItemChannelLinkProvider.getAll();
         for (ItemChannelLink link : links) {
-            if (link.getChannelUID().equals(channelUID)) {
+            if (link.getUID().equals(channelUID)) {
                 logger.info(
                         "Channel '{}' is already linked to item '{}' and will be unlinked before it will be linked to the new item.",
```

Comment:
would change it to debug as it looks it can regularly happen during normal operation?